### PR TITLE
create cart context and wrap around app

### PR DESCRIPTION
**Pull request by isapind**
# Create Cart Context

**Ticket:** [Create Cart Context](https://trello.com/c/w8iuZiQC/76-create-cart-context)

## Description

Created a context for cart that wraps around the app component. It currently has a hard coded state, taken from the trello ticket, and which we can see using React Developer Tools

### Screenshots

Screenshot 2022-05-12 at 17 08 03

## Effected Areas

App, added a context folder in src.

## Developer Checklist

I have

- [x] Made the minimum amount of change required to achieve my goal to a high standard
- [ ] Added tests that describe the change
- [ ] Created tickets for any tech debt incurred

### CODE REVIEW

File: src/contexts/Cart.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React, { useState } from 'react';
+
+type CartItems = { id: string; name: string; quantity: number };
+type Cart = CartItems[];
+type CartState = {
+  cart: Cart;
+  setCart: React.Dispatch<React.SetStateAction<Cart>>;
+};
+type CartProviderProps = { children: React.ReactNode };
```
Comment (Si3rr4wow) [inline src/contexts/Cart.tsx:9]: you don't need this type, `React.FC` knows about children

File: src/contexts/Cart.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      id: '62617159f21bcb5fa89e67ab',
+      name: 'Think Personal Front',
+      quantity: 3,
+    },
+    {
+      id: '62617159f21bcb5fa89e67ac',
+      name: 'Ready',
+      quantity: 1,
+    },
+  ];
```
Comment (Si3rr4wow) [inline src/contexts/Cart.tsx:27]: if you move this up and out of the component then you can feed it straight into the `createContext` then the context type doesn't have to have undefined in it

**Review by Si3rr4wow — APPROVED** [review]
nailed it!

**Review by KhalidStonehouse2001 — APPROVED** [review]
Great Work!

**Pull request merged**